### PR TITLE
[Ubuntu] Migrate Toolset tests to Pester

### PR DESCRIPTION
**Pull request by dibir-magomedsaygitov**
# Description
In scope of this PR we are migrating tests for Toolset tools to Pester.

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by al-cheb — APPROVED** [review]

File: images/linux/scripts/installers/Install-Toolset.ps1 [hunk 1]
```diff
@@ -55,4 +57,6 @@ foreach ($tool in $tools) {
 
 chown -R "$($env:SUDO_USER):$($env:SUDO_USER)" /opt/hostedtoolcache/Python
 chown -R "$($env:SUDO_USER):$($env:SUDO_USER)" /opt/hostedtoolcache/node
-chown -R "$($env:SUDO_USER):$($env:SUDO_USER)" /opt/hostedtoolcache/go
\ No newline at end of file
+chown -R "$($env:SUDO_USER):$($env:SUDO_USER)" /opt/hostedtoolcache/go
+
+Invoke-PesterTests -TestFile "Toolset" -TestName "Toolset"
```
Comment (vsafonkin) [inline images/linux/scripts/installers/Install-Toolset.ps1:62]: I would suggest to run tests in `Configure-Toolset.ps1`

Comment (AlenaSviridenko) [reply images/linux/scripts/installers/Install-Toolset.ps1:62]: And no need to specify TestName here, it is the only test suite in the file

File: images/linux/scripts/tests/Toolset.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,69 @@
+Describe "Toolset" {
+    $tools = (Get-ToolsetContent).toolcache
```
Comment (AlenaSviridenko) [inline images/linux/scripts/tests/Toolset.Tests.ps1:2]: so in fact we are testing toolcache here. Maybe makes sense to rename file and test suite to Toolcache?

**Review by dmitry-shibanov — APPROVED** [review]

File: images/linux/scripts/installers/Install-Toolset.ps1 (hunk 1 above)
Comment (dibir-magomedsaygitov) [reply images/linux/scripts/installers/Install-Toolset.ps1:62]: Thank you for your suggestions. I would prefer to leave it as it is. We install tools in this script, so it make sense to invoke tests from here as well.

File: images/linux/scripts/tests/Toolset.Tests.ps1 (hunk 2 above)
Comment (dibir-magomedsaygitov) [reply images/linux/scripts/tests/Toolset.Tests.ps1:2]: Thank you for your point, @AlenaSviridenko. I'd prefer to leave current naming as it is for now. Just to be consistent with other OS's.
cc: @maxim-lobanov

File: images/linux/scripts/installers/Install-Toolset.ps1 (hunk 1 above)
Comment (vsafonkin) [reply images/linux/scripts/installers/Install-Toolset.ps1:62]: But earlier we had actions: `install` -> `configure` -> `validate`. Now, this chain will be changed to `install` -> `validate (run tests)`-> `configure`.

Comment (dibir-magomedsaygitov) [reply images/linux/scripts/installers/Install-Toolset.ps1]: Good point, @vsafonkin, thank you! Moved test invocation to `Configure-Toolset.ps1`.

**Review by vsafonkin — APPROVED** [review]

File: images/linux/scripts/tests/Toolset.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                        }
+                    }
+                }
+
+                $foundVersionName = $foundVersion.name
+                if ($tool.name -eq 'PyPy')
+                {
+                    $pypyVersion = & "$foundVersionPath/bin/python" -c "import sys;print(sys.version.split('\n')[1])"
+                    $foundVersionName = "{0} {1}" -f $foundVersionName, $pypyVersion
+                }
```
Comment (maxim-lobanov) [inline images/linux/scripts/tests/Toolset.Tests.ps1:65]: Let's remove these lines. Seems redundant

File: images/linux/scripts/tests/Toolset.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                    | Sort-Object -Property {[version]$_.name} -Descending `
+                    | Select-Object -First 1
+                $foundVersionPath = Join-Path $foundVersion $tool.arch
+
+                if($toolExecs) {
+                    foreach ($executable in $toolExecs["tools"]) {
+                        $executablePath = Join-Path $foundVersionPath $executable   
+    
+                        It "Validate $executable" -TestCases @{ExecutablePath = $executablePath} {
+                            $ExecutablePath | Should -Exist
```
Comment (maxim-lobanov) [inline images/linux/scripts/tests/Toolset.Tests.ps1:55]: nitpick: spaces

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**